### PR TITLE
Purchases: Standardize name of privacy protection upgrade

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request fixes https://github.com/Automattic/wp-calypso/issues/1938 by renaming all occurrences of _private registration_ into _privacy protection_ in order to be consistent and reduce confusion, both for users and for us developers. 

#### Testing instructions

You should check that _private registration_ is never mentioned anywhere in the following flow:

1. Run `git checkout update/privacy-protection` and start your server, or open a [live branch](https://calypso.live/?branch=update/privacy-protection)
2. Open the [`Home` page](http://calypso.localhost:3000/) and log in
3. Sandbox the store
4. Navigate to the [`Domains` page](http://calypso.localhost:3000/domains/add/stephanethomas222.wordpress.com)
5. Purchase a plan with a domain registration and privacy protection
6. Purchase another domain without privacy protection this time
7. Navigate to the `Domains` page again
8. Click the second domain registration
9. Click the `None` flag in front of `Privacy Protection:`
10. Click the `Enable Privacy Protection` link
11. Click the `Add Privacy Protection` button
12. Navigate to the [`Purchases` page](http://calypso.localhost:3000/me/purchases) now
13. Click the domain registration with privacy protection
14. Click the `Cancel Privacy Protection` link
15. Click the `Cancel Privacy Protection` button

#### Additional notes
 
The following occurrences have not been renamed because this would involve updating the API:

* API properties:
[`has_private_registration`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/lib/domains/assembler.js#L30)
[`has_private_registration`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/lib/purchases/assembler.js#L26)
[`has_private_registration`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/state/sites/domains/assembler.js#L18)

* API error codes:
[`enable_private_reg_failed`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/my-sites/upgrades/domain-management/transfer/transfer-out/unlocked.jsx#L41)
[`unlock_domain_and_disable_private_reg_failed` and `unlock_domain_failed`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/my-sites/upgrades/domain-management/transfer/transfer-out/shared.js#L15-L24)

* Tests and fixtures:
[`has_private_registration`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/state/purchases/test/reducer.js#L205-L217)
[`has_private_registration`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/state/sites/domains/test/fixture.js#L102-L134)

Finally the [`cancel_private_registration`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/me/purchases/cancel-private-registration/index.jsx#L53-L59) stat has not been renamed because this would break history.

#### Reviews
 
- [x] Code
- [x] Product

### CODE REVIEW

File: client/me/purchases/titles.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import i18n from 'i18n-calypso';
 
 export default {
 	addCreditCard: i18n.translate( 'Add Credit Card' ),
-	cancelPrivateRegistration: i18n.translate( 'Cancel Private Registration' ),
+	cancelPrivateRegistration: i18n.translate( 'Cancel Privacy Protection' ),
```
Comment (a8ci18n) [inline client/me/purchases/titles.js]: Alternate string suggestion: 
* `translate( 'Privacy Protection' )`&mdash; translations: **21**. *ES Score: 7.32*

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/privacy-protection)

File: client/me/purchases/cancel-private-registration/index.jsx [hunk 2]
```diff
@@ -90,7 +90,7 @@ const CancelPrivateRegistration = React.createClass( {
 		this.props.cancelPrivateRegistration( id ).then( () => {
 			this.resetState();
 
-			notices.success( this.translate( 'You have successfully canceled private registration for %(domain)s.', {
+			notices.success( this.translate( 'You have successfully canceled privacy protection for %(domain)s.', {
```
Comment (a8ci18n) [inline client/me/purchases/cancel-private-registration/index.jsx]: Warning: 20 translations will be lost with this change.
Alternate string suggestion: 
* `translate( 'You have successfully canceled the Private Registration upgrade for your domain.' )`&mdash; translations: **23**. *ES Score: 5.23*

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/privacy-protection)

File: client/me/purchases/titles.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ import i18n from 'i18n-calypso';
 
 export default {
 	addCreditCard: i18n.translate( 'Add Credit Card' ),
-	cancelPrivateRegistration: i18n.translate( 'Cancel Private Registration' ),
+	cancelPrivacyProtection: i18n.translate( 'Cancel Privacy Protection' ),
```
Comment (a8ci18n) [inline client/me/purchases/titles.js]: Alternate string suggestion: 
* `translate( 'Privacy Protection' )`&mdash; translations: **21**. *ES Score: 7.32*

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/privacy-protection)

File: client/me/purchases/cancel-privacy-protection/index.jsx [hunk 4]
```diff
@@ -87,10 +87,10 @@ const CancelPrivateRegistration = React.createClass( {
 			cancelling: true
 		} );
 
-		this.props.cancelPrivateRegistration( id ).then( () => {
+		this.props.cancelPrivacyProtection( id ).then( () => {
 			this.resetState();
 
-			notices.success( this.translate( 'You have successfully canceled private registration for %(domain)s.', {
+			notices.success( this.translate( 'You have successfully canceled privacy protection for %(domain)s.', {
```
Comment (a8ci18n) [inline client/me/purchases/cancel-privacy-protection/index.jsx]: Warning: 20 translations will be lost with this change.
Alternate string suggestion: 
* `translate( 'You have successfully canceled the Private Registration upgrade for your domain.' )`&mdash; translations: **23**. *ES Score: 5.23*

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/privacy-protection)

**Comment by drewblaisdell** [comment]
The changes look good to me.

> The following occurrences have not been renamed because this would involve updating the API:

What do you think about updating the product name that the API returns for privacy protection? It's a little strange to have this in the sidebar when checking out:

screen shot 2017-01-27 at 12 28 46 pm

when every other reference uses "privacy protection" now. I think this should be a trivial API change, but if I'm wrong, then we should just deploy this PR because it is still more consistent than what we currently have.

**Comment by aidvu** [comment]
LGTM. Would you mind also updating https://github.com/Automattic/wp-calypso/blob/e3e7f314c7ee7c75cd319a92e416cf8b834d2ce8/client/me/purchases/cancel-private-registration/index.jsx#L114 to say `privacy protection` upgrade?

**Comment by stephanethomas** [comment]
> Feel free to merge after updating the cancel private upgrade text.

Good point. I've just updated this string and rebased.

**Comment by stephanethomas** [comment]
> What do you think about updating the product name that the API returns for privacy protection?

I think it makes absolutely sense. I was only focusing on the client part in this pull request but I'll take care of the API and emails in a server patch.

**Pull request merged**